openff/toolkit/topology/topology.py

```diff
@@ -885,14 +958,60 @@ def topology_particle_index(self):
         # This assumes that the particles in a topology are listed with all atoms from all TopologyMolecules
         # first, followed by all VirtualSites from all TopologyMolecules second
         orientation_key = self._virtual_particle.orientation
-        offset = 0
+        offset = None
         # vsite is a topology vsite, which has a regular vsite
-        for i, ornt in enumerate(self._virtual_site._virtual_site.orientations):
+        for i, ornt in enumerate(self._virtual_site.orientations):
+
             if ornt == orientation_key:
                 offset = i
                 break
 
-        return offset + self._virtual_site.topology_virtual_particle_start_index
+        assert offset is not None
```

Comment:
Add a safety check to make sure we aren't returning a value when the orientation has not been found